Uglify is going in infinite loop starting from v3.5.0

**Bug report or feature request?**
Starting from v 3.5.0, it looks like the library goes in infinite loop

<!-- Note: sub-optimal but correct code is not a bug -->

**Uglify version (`uglifyjs -V`)**
3.5.0
**JavaScript output or error produced.**

Plumber found unhandled error:
 GulpUglifyError: unable to minify JavaScript
Caused by: RangeError: Maximum call stack size exceeded
File: /d/projects/citelighter/web-app/assets/gulp/js/admin_cms_lessons.min.js

--- a/lib/compress.js
+++ b/lib/compress.js
@@ -2246,7 +2246,11 @@ merge(Compressor.prototype, {
         });
         def(AST_SymbolRef, function() {
             var fixed = this.fixed_value();
-            return fixed && fixed.is_truthy();
+            if (!fixed) return false;
+            this.is_truthy = return_false;
+            var result = fixed.is_truthy();
+            delete this.is_truthy;
+            return result;
         });
     })(function(node, func) {
         node.DEFMETHOD("is_truthy", func);
@@ -2300,7 +2304,11 @@ merge(Compressor.prototype, {
             if (is_undeclared_ref(this) && this.is_declared(compressor)) return false;
             if (this.is_immutable()) return false;
             var fixed = this.fixed_value();
-            return !fixed || fixed._dot_throw(compressor);
+            if (!fixed) return true;
+            this._dot_throw = return_true;
+            var result = fixed._dot_throw(compressor);
+            delete this._dot_throw;
+            return result;
         });
         def(AST_UnaryPrefix, function() {
             return this.operator == "void";
@@ -2342,7 +2350,11 @@ merge(Compressor.prototype, {
         });
         def(AST_SymbolRef, function(compressor) {
             var fixed = this.fixed_value();
-            return fixed && fixed.is_boolean(compressor);
+            if (!fixed) return false;
+            this.is_boolean = return_false;
+            var result = fixed.is_boolean(compressor);
+            delete this.is_boolean;
+            return result;
         });
         var unary = makePredicate("! delete");
         def(AST_UnaryPrefix, function() {
@@ -2427,7 +2439,11 @@ merge(Compressor.prototype, {
         });
         def(AST_SymbolRef, function(compressor) {
             var fixed = this.fixed_value();
-            return fixed && fixed.is_number(compressor);
+            if (!fixed) return false;
+            this.is_number = return_false;
+            var result = fixed.is_number(compressor);
+            delete this.is_number;
+            return result;
         });
         var unary = makePredicate("+ - ~ ++ --");
         def(AST_Unary, function() {
@@ -2456,7 +2472,11 @@ merge(Compressor.prototype, {
         def(AST_String, return_true);
         def(AST_SymbolRef, function(compressor) {
             var fixed = this.fixed_value();
-            return fixed && fixed.is_string(compressor);
+            if (!fixed) return false;
+            this.is_string = return_false;
+            var result = fixed.is_string(compressor);
+            delete this.is_string;
+            return result;
         });
         def(AST_UnaryPrefix, function() {
             return this.operator == "typeof";

